src/app/farming/farming.service.ts

```diff
@@ -134,31 +134,35 @@ async function farmItems(store: D1Store) {
 // Ensure that there's one open space in each category that could
 // hold an item, so they don't go to the postmaster.
 async function makeRoomForItems(store: D1Store) {
+  const buckets = await getBuckets();
+  const makeRoomBuckets = makeRoomTypes.map((type) => buckets.byId[type]);
+  makeRoomForItemsInBuckets(store, makeRoomBuckets, D1StoresService);
+}
+
+// Ensure that there's one open space in each category that could
+// hold an item, so they don't go to the postmaster.
+export async function makeRoomForItemsInBuckets(
+  store: DimStore,
+  makeRoomBuckets: InventoryBucket[],
+  storeService: StoreServiceType
+) {
   // If any category is full, we'll move one aside
-  const itemsToMove: D1Item[] = [];
+  const itemsToMove: DimItem[] = [];
+  const itemInfos = rxStore.getState().inventory.itemInfos;
   makeRoomTypes.forEach((makeRoomType) => {
     const items = store.buckets[makeRoomType];
     if (items.length > 0 && items.length >= store.capacityForItem(items[0])) {
```

Comment:
as soon as i start farming mode, i get
```
Uncaught TypeError: Cannot read property 'length' of undefined
    at farming.service.ts:154
    at Array.forEach (<anonymous>)
    at makeRoomForItemsInBuckets (farming.service.ts:152)
    at makeRoomForItems (d2farming.service.ts:93)
```
on that first `items.length`

Reply:
`makeRoomType` is `BUCKET_PRIMARY_WEAPON`
and `store.buckets` has hashes for keys

Reply:
this looks like it's running code intended for d1

Reply:
Thanks, I fixed it.